### ISSUE TITLE
[Improvement]: Introduce a temporary folder for unit test of AMS server

### ISSUE THREAD

**Issue by wangtaohz** [NetEase/arctic#1310]
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

If we run the unit test of AMS server, the local `/tmp` folder will be used, and it will be cleaned before test, also there will be a `derby` folder in the project ,like `ams/ams-server/src/test/java/com/netease/arctic/ams/server/derby/`.

I think all these folders should be in a temporary folder.

### How should we improve?

Introduce temporary folder for unit test of AMS server.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wangtaohz** [comment]
The related code is here, in `com.netease.arctic.ams.server.AmsTestBase`

```
public class AmsTestBase {

  private static final File testTableBaseDir = new File("/tmp");
  private static final File testBaseDir = new File("unit_test_base_tmp");
  public static ArcticTableMetastoreHandler amsHandler;
  protected static HMSMockServer hms;
  public static ArcticCatalog catalog;
  public static ArcticCatalog icebergCatalog;
  public static final String AMS_TEST_CATALOG_NAME = "ams_test_catalog";

  public static final String AMS_TEST_DB_NAME = "ams_test_db";
  public static final String CATALOG_CONTROLLER_UNITTEST_NAME = "unit_test";

  public static final String AMS_TEST_ICEBERG_CATALOG_NAME = "ams_test_iceberg_catalog";
  public static final String AMS_TEST_ICEBERG_DB_NAME = "ams_test_iceberg_db";

  @ClassRule
  public static final TemporaryFolder tempFolder = new TemporaryFolder();

  @BeforeClass
  public static void beforeAllTest() throws Exception {
    System.setProperty("HADOOP_USER_NAME", System.getProperty("user.name"));
    FileUtils.deleteQuietly(testBaseDir);
    FileUtils.deleteQuietly(testTableBaseDir);
    tempFolder.create();
    testBaseDir.mkdirs();
    testTableBaseDir.mkdir();

```

**Comment by wangtaohz** [comment]
The issue has been fixed in version 0.5.0
